experimental/dds/tree/src/SharedTree.ts

```diff
@@ -1549,51 +1564,114 @@ export class SharedTree extends SharedObject<ISharedTreeEvents> implements NodeI
 	 *
 	 * @param content - op to apply locally.
 	 */
-	protected applyStashedOp(op: unknown): void {
+	protected applyStashedOp(op: unknown): StashedLocalOpMetadata {
+		// In some scenarios, edit ops need to have their edits transformed before application and resubmission. The transformation
+		// occurs in this method, and the result is passed to `resubmitCore` via the return value of this function.
 		const sharedTreeOp = op as SharedTreeOp | SharedTreeOp_0_0_2;
 		switch (sharedTreeOp.type) {
 			case SharedTreeOpType.Edit: {
+				let stashedEdit: Edit<ChangeInternal> | undefined;
 				switch (this.writeFormat) {
 					case WriteFormat.v0_0_2:
 						switch (sharedTreeOp.version) {
 							case WriteFormat.v0_0_2: {
-								const edit = this.parseSequencedEdit(sharedTreeOp);
-								this.applyEditLocally(edit, undefined);
+								stashedEdit = this.parseSequencedEdit(sharedTreeOp);
 								break;
 							}
 							case WriteFormat.v0_1_1:
-								// TODO:#74390: Implement
 								fail('Received stashed op 0.1.1 before upgrade');
 							default:
 								fail('Unknown version');
 						}
 						break;
 					case WriteFormat.v0_1_1:
 						switch (sharedTreeOp.version) {
-							case WriteFormat.v0_0_2:
-								// TODO:#74390: Implement
-								fail('v0.1.1 does not support stashed ops.');
-							case WriteFormat.v0_1_1:
-								// TODO:#74390: Implement
-								fail('v0.1.1 does not support stashed ops.');
+							case WriteFormat.v0_0_2: {
+								// Use the IDs from the stashed ops as overrides for the equivalent new ops
+								stashedEdit = convertEditIds(sharedTreeOp.edit, (id) => this.generateNodeId(id));
+								break;
+							}
+							case WriteFormat.v0_1_1: {
+								assert(this.stashedIdCompressor !== null, 'Stashed op applied after expected window');
+								if (this.stashedIdCompressor === undefined) {
+									// Use a temporary compressor that will help translate the stashed ops
+									this.stashedIdCompressor = IdCompressor.deserialize(
+										this.idCompressor.serialize(false),
+										stashedSessionId,
+										sharedTreeOp.idRange.attributionId
+									);
+									// Once all stashed ops have been applied, clear the temporary state
+									this.runtime.on('connected', () => {
```

Comment:
Are there no interleavings that can apply a stashed op after this event fires? I'm not too familiar :)

Reply:
My understanding is that the container is fully connected once it has finished "catching up" (both trailing ops and stashed ops). @wes-carlson recommended taking this approach, feel free to chime in if there is more to it than that.